### DISCUSSION TITLE
Query by case insensitive regex

### DISCUSSION THREAD

**Discussion by guyr22** [art049/odmantic#446] [Q&A]
Hi, im changing my project from working with pymongo to odmantic and i cant find how can i execute this query
db.find_one({"name": {"$regex": f"^{name}$", "$options": "i"}})
I execute this query in a function where i get the parameter "name" and i want to check if it exists in my db but i want to check it case-insensitively.

**Comment by mbezhanov** [comment]
Let's say your model name is `Foo`. Use this:

```python
regex = re.compile(f"^{name}$", re.IGNORECASE)
query = Foo.name.match(regex)
results = await engine.find(Foo, query)
```

The idea is to compile a case-insensitive regular expression in Python first, and then pass this pre-compiled expression to the `match()` method. Odmantic will handle the rest.
